rust/lance-index/src/scalar/inverted.rs

```diff
@@ -0,0 +1,513 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use arrow::array::{AsArray, ListBuilder, UInt64Builder};
+use arrow::datatypes;
+use arrow_array::{ArrayRef, RecordBatch, StringArray, UInt32Array, UInt64Array};
+use arrow_schema::{DataType, Field};
+use async_trait::async_trait;
+use datafusion::execution::SendableRecordBatchStream;
+use deepsize::DeepSizeOf;
+use futures::{StreamExt, TryStreamExt};
+use itertools::Itertools;
+use lance_core::{Error, Result, ROW_ID};
+use roaring::RoaringBitmap;
+use snafu::{location, Location};
+
+use crate::Index;
+
+use super::{IndexReader, IndexStore, ScalarIndex, ScalarQuery};
+
+const TOKENS_FILE: &str = "tokens.lance";
+const INVERT_LIST_FILE: &str = "invert.lance";
+const DOCS_FILE: &str = "docs.lance";
+
+const TOKEN_COL: &str = "_token";
+const TOKEN_ID_COL: &str = "_token_id";
+const FREQUENCY_COL: &str = "_frequency";
+const NUM_TOKEN_COL: &str = "_num_tokens";
+
+#[derive(Debug, Clone, Default, DeepSizeOf)]
+pub struct InvertedIndex {
+    tokens: TokenSet,
+    invert_list: InvertedList,
+    docs: DocSet,
```

Comment:
Can you describe a little what these things are?  Either here or in the structs?

Reply:
added